code/launcher/xmake.lua

```diff
@@ -7,22 +7,36 @@ target("Launcher")
             local stdout, stderr = os.iorunv(cmd, args)
         end
 
+        local function run_install(cmd)
+            if os.getenv("IS_CI") == "true" then
+                run_command(cmd, {"ci"})
+            else
+                run_command(cmd, {"install"})
+            end
+        end
+
         os.cd("code/launcher")
 
         local npm = is_host("windows") and "npm.cmd" or "npm"
 
         -- Run npm commands in the root directory
-        run_command(npm, {"install"})
-        run_command(npm, {"run", "build:all"})
+        run_install(npm)
+        run_command(npm, {"run", "build"})
 
         local electron_dir = path.join(os.curdir(), "electron")
         if not os.isdir(electron_dir) then
             raise("Electron directory not found: " .. electron_dir)
         end
 
+        os.cd(electron_dir)
+
+        -- Run npm commands in the electron directory
+        run_install(npm)
+        run_command(npm, {"run", "build:electron"})
+
         -- Copy produced files
         local src = path.join(electron_dir, "out", "win-unpacked")
-        local dst = path.join("..", "..", target:installdir("launcher"))
+        local dst = path.join("..", "..", "..", target:installdir("launcher"))
```

Comment:
ah it's back ?

Reply:
Yes, like said in previous PR.
In order to use `npm ci` for cache purpose, it needs to be run in `launcher/` and `launcher/electron/`. But `npm ci` is only used when `IS_CI=true` (for workflow). So in the end it is simpler to just keep the original behavior of running `npm install` in two different paths. So it works fine without messy script command / trouble between workflow / dev / prod environments.
`cd` is back, therefore `..` is back too.